alacritty/src/event.rs

```diff
@@ -227,17 +229,17 @@ impl<'a, N: Notify + 'a, T: EventListener> input::ActionContext<T> for ActionCon
     }
 
     fn change_font_size(&mut self, delta: f32) {
-        self.resize_pending.font_size = Some(FontResize::Delta(delta));
+        self.display_update_pending.font = Some(FontUpdate::SizeDelta(delta));
         self.terminal.dirty = true;
     }
 
     fn reset_font_size(&mut self) {
-        self.resize_pending.font_size = Some(FontResize::Reset);
+        self.display_update_pending.font = Some(FontUpdate::ResetToConfig);
```

Comment:
@chrisduerr  just FYI, this is where I found it necessary to distinguish between `ResetTo(Font)` and `ResetToConfig`; at this point in the `ActionContext` we don't have access to the `Font` to which we want to reset. We could get more specific here in the enum value's name, ie `ResetToProcessorConfigFont`, what do you think?

Reply:
> at this point in the ActionContext we don't have access to the Font to which we want to reset

In theory it would be possible to modify the font though. We can just store `&'a mut Font` in the context, modify it in here, then stage the latest version of the font in the `DisplayUpdate`.

Reply:
I didn't see a good way to use `&'a mut Font` since the font we are interested comes from a `&'a mut Config` when initializing the `ActionContext` just before draining the event queue; I also didn't see a good way to work around that `Config` being mutable since it needs to be replaced whenever the config file is reloaded. Because of this I instead [cloned the `Font` onto the `ActionContext`](https://github.com/jwilm/alacritty/pull/2836/commits/2a7d15ee590efee8024607168228dfda8ffd0b75) to avoid the `ResetToConfig` variant.

I think this does introduce an unlikely race condition where the font size reset command is issued around the exact same time the config file is reloaded such that the two events land on the queue at the same time in which case the font size reset event may end up resetting to the old version of the font size (assuming my understanding of the way event processing works is adequate).